### ISSUE TITLE
Add an "Export Selected Audio" to the Extra>Export menu AND/OR enable parameter setting for Export Audio when used in a Macro

### ISSUE THREAD

**Issue by petersampsonaudacity** [audacity/audacity#5581]
### Your idea
Add an "Export Selected Audio" to the Extra>Export menu:
![image](https://github.com/audacity/audacity/assets/50205984/ca3ff4ca-e5c4-47e7-bdf1-2a4d812067a1)

This will enable users who used to be able to have a shortcut to this to aid their workflow.

Note that this was possible in earlier versions of Audacity before the new Export dialog was introduced in 3.4 as there was an explicit **File > Export > Export Selected Audio** command.

I can understand why the designers do not want to clutter the File menu/submenus - hence the suggestion to place it out of the way in **Extra > Export**

### Problem to be solved
With 3.4 and onwards there is no way of setting a shortcut for export selected audio.  It appears that VO (voiceover) users seem to value this highly.

Originally reported on the Forum:
https://forum.audacityteam.org/t/please-bring-back-the-export-selected-audio-command/89362

### Prior art
![image](https://github.com/audacity/audacity/assets/50205984/fbf92c28-0864-4adc-b964-f2ce6d033265)

![image](https://github.com/audacity/audacity/assets/50205984/6b6b9deb-cdf3-4ac0-afc9-5e9792b7bb1a)

### Additional context
Technically this is a regression as although export of selected audio is still available via the new export dialog, the user can no longer set a direct shortcut for Export Selected Audio.

**Comment by petersampsonaudacity** [comment]
@LWinterberg @crsib 

Might this be better addressed if the user could specify parameters (including Export Range) in a macro - as one can with Effects in a macro?

Then users that want to Export selected don't need a command they can just build a macro with the normal Export Audio command.

Is this something that is possible to do, or is there something in the app that blocks parameter setting for Export Audio in a Macro?

**Comment by bokeron2020** [comment]
I don't know who was the fucking genious that decided to remove a very useful feature just because it looked ugly. Or the fucking geniuses who allowed it.
Since acquisition this program is worst with every iteration.

**Comment by petersampsonaudacity** [comment]
@LWinterberg 
I'm assuming that if and when Export Selected Audio appears in the Extra>Export sub-menu that it will also show in the list of available macro export commends.

Will this be the case?

**Comment by petersampsonaudacity** [comment]
@LWinterberg  Another request for this on the Forum today:  https://forum.audacityteam.org/t/3-5-beta-and-export-selected-audio/102850

I note, Leo, that you still have this listed as part of your "dejankification" - any chance we will see this in 3.5.0 please?
 See:   **Cleanup tasks #5710**

**Comment by dozzzzer** [comment]
Tested

**Comment by petersampsonaudacity** [comment]
Tested this on W10 with @vsverchinsky 's branch build:  audacity-win-3.6.0-alpha-20240405+6ac9f75-x64

1.  the command is present in the Extra>Export submenu
2. the export works correctly, displaying the Export Audio with **Current Selection** pre-selected
3. the last exported to format is remembered and reused on next use of the command both with Audacity remaining active or on next launch
4. I was able to successfully create a shortcut for the command.
